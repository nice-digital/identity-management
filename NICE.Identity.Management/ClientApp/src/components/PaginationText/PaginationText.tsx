import React, { FC } from "react";
import { useListInfo } from "src/helpers/useListInfo";

type PaginationTextProps = {
	dataLength: number;
	labelSingular: string;
	labelPlural: string;
};

export const PaginationText: FC<PaginationTextProps> = ({
	dataLength,
	labelSingular,
	labelPlural,
}) => {
	const { paginationStart, paginationFinish, itemsPerPage } =
		useListInfo(dataLength);

	return (
		<h2
			className="results-heading"
			data-qa-sel={`${labelPlural}-returned`}
		>
			{`Showing ${
				dataLength > itemsPerPage
					? `${paginationStart} to ${paginationFinish} of `
					: ""
			}${dataLength} ${
				dataLength === 1 ? `${labelSingular}` : `${labelPlural}`
			}`}
		</h2>
	);
};
