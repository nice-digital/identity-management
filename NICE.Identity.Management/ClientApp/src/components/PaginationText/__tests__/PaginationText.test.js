import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import toJson from "enzyme-to-json";
import { nextTick } from "../../../utils/nextTick";
import { PaginationText } from "../PaginationText";

describe("PaginationText", () => {
    const paginationTextOrgProps = {
        dataLength: 32,
	    labelSingular: "organisation",
	    labelPlural: "organisations",
	};

	it("should match the snapshot when pagination is default and organisations props passed in", async () => {
		const wrapper = mount(
			<MemoryRouter>
				<PaginationText {...paginationTextOrgProps} />
			</MemoryRouter>,
		);
		await nextTick();
		wrapper.update();
		expect(toJson(wrapper, { noKey: true, mode: "deep" })).toMatchSnapshot();
	});

    it("should match the snapshot when on page 2 and organisations props passed in", async () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={['/organisations?page=2']}>
				<PaginationText {...paginationTextOrgProps} />
			</MemoryRouter>,
		);
		await nextTick();
		wrapper.update();
		expect(toJson(wrapper, { noKey: true, mode: "deep" })).toMatchSnapshot();
	});
});