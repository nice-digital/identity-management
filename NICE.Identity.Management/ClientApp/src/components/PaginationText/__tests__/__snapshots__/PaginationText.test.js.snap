// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot when on page 2 and organisations props passed in 1`] = `
<div>
  <h2
    class="h5 mv--0"
    data-qa-sel="organisations-returned"
  >
    Showing 26 to 32 of 32 organisations
  </h2>
</div>
`;

exports[`should match the snapshot when pagination is default and organisations props passed in 1`] = `
<div>
  <h2
    class="h5 mv--0"
    data-qa-sel="organisations-returned"
  >
    Showing 1 to 25 of 32 organisations
  </h2>
</div>
`;
